modules/api/php/provisioners/projectdicomsrowprovisioner.class.inc

```diff
@@ -0,0 +1,100 @@
+<?php declare(strict_types=1);
+/**
+ * This file implements a data provisioner to get all dicoms of a project
+ * created since a given date.
+ *
+ * PHP Version 7
+ *
+ * @category API
+ * @package  Loris
+ * @author   Xavier Lecours Boucher <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\api\Provisioners;
+
+use \LORIS\Data\Provisioners\DBObjectProvisioner;
+use \LORIS\api\Models\ProjectDicomsRow;
+/**
+ * This file implements a data provisioner to get all dicoms of a project
+ * created since a given date.
+ *
+ * PHP Version 7
+ *
+ * @category API
+ * @package  Loris
+ * @author   Xavier Lecours Boucher <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class ProjectDicomsRowProvisioner extends DBObjectProvisioner
+{
+    /**
+     * Create a RowProvisioner
+     *
+     * @param \Project  $project   The project from which dicoms are requested
+     * @param \DateTime $since     The date from which dicoms are requested
+     * @param string    $classname The class name of the returned objects
+     */
+    function __construct(
+        \Project $project,
+        \DateTime $since,
+        string $classname='default'
+    ) {
+        parent::__construct(
+            '
+             SELECT
+               s.CandID as Candidate,
+               c.PSCID as PSCID,
+               c.Entity_type as Entity_type,
+               s.Visit_label as Visit,
+               s.Date_visit as Visit_date,
+               s.CenterID as CenterID,
+               p.Name as Site,
+               t.DateAcquired as date_acquired,
+               t.DateFirstArchived as date_first_archived,
+               t.DateLastArchived as date_last_archived,
+               t.TarchiveID as tarchiveid,
+               t.DicomArchiveID as DicomArchiveID,
+               t.ArchiveLocation as Archive,
+	       t.SourceLocation as Source,
+             FROM
+               tarchive t
+             LEFT JOIN  session s
+               ON (t.SessionID = s.ID)
```

Comment:
there are NULL values in the table `tarchive` in raisinbread, is this normal?

Reply:
which columns?  
@spell00 can you pinpoint the rows you think might be problematic and we'll take a look? 
please link to the sql lines in raisinbread on aces/loris for common reference.

Reply:
Yes, my guess is these are the few that I had to add in the raisinbread where the PSCID and CandID in the PatientName did not refer to an existing candidate (hence, no MINC images inserted into the files table).

Basically, because of the bad PatientName present in the DICOMs, the pipeline cannot associate them to a given candidate/session and therefore their SessionID is NULL. I am not worried about those cases as typically, projects would end up removing those entries, relabelling the DICOMs correctly and reupload them with the correct PatientName.

Hope that makes sense. @spell00 let me know if that is not clear.